src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -242,9 +246,13 @@ public override byte[] ToBinary(object obj)
         /// <returns>The object contained in the array</returns>
         public override object FromBinary(byte[] bytes, Type type)
         {
-            string data = Encoding.UTF8.GetString(bytes);
-            object res = JsonConvert.DeserializeObject(data, Settings);
-            return TranslateSurrogate(res, this, type);
+            using (var stream = new MemoryStream(bytes))
+            using (var r = new StreamReader(stream))
+            using (var reader = new JsonTextReader(r))
+            {
+                object res = _serializer.Deserialize(reader);
```

Comment:
We cannot trust the Thread safety of `JsonSerializer` in it's present state. You either have to create a new Serializer each time or use the static methods.

Reply:
Dang, that's disappointing. Why does the current implementation hold on to a reference of an instance of the serializer then?

Reply:
> Dang, that's disappointing. Why does the current implementation hold on to a reference of an instance of the serializer then?

No idea; we probably don't need to have it anymore. Although having a `ThreadStatic` reference to it might not be the worst thing in the world, in combination with your changes.